File: src/iterator/java/com/my/UserList.java
package src.iterator.java.com.my;

import src.bridge.java.com.my.abstraction.Message;
import src.prototype.java.com.my.db.model.User;

public class UserList {
    private final Node head;

    public UserList(Node head) {
        this.head = head;
    }

    public Node iterator() {
        return head;
    }

    public static class Node implements Iterator {
        private final User current;
        private final Node next;

        public Node(User current, Node next) {
            this.current = current;
            this.next = next;
        }

        public boolean hasNext() {
            return current != null;
        }

        @Override
        public String toString() {
            return "Node{" +
                    "current=" + current +
                    '}';
        }

        public Node getNext() {
            if (hasNext()) {
                return next;
            }

            return this;
        }
    }
}

File: src/iterator/python/UserList.py
from src.iterator.python.Iterator import Iterator
from src.prototype.python.db.model.User import User

class Node(Iterator):
    def __init__(self, current, n):
        self.current = current
        self.next = n

    def has_next(self):
        return self.current is not None

    def get_next(self):
        if self.has_next():
            return self.next

        return self

    def __str__(self):
        return ("Node{" +
                "current=" + str(self.current) +
                '}')


class UserList:
    head: Node = None

    def __init__(self, head: Node):
        self.head = head

    def iterator(self):
        return self.head
